### DISCUSSION TITLE
Edge reconnect tool support

### DISCUSSION THREAD

**Discussion by ionutpredoaia** [eclipse-sirius/sirius-components#3470] [Q&A]
Hello,

Does Sirius Web support the reconnection of edges as in Sirius Desktop? Can a reconnect tool be defined at runtime in Sirius Web, or alternatively, could a reconnect tool be defined programmatically via some existing APIs?

Thanks,
Ionut

**Comment by pcdavid** [comment]
Yes this is supported.

* Inside the _Edge Description_ for which you want to support reconnection, create an _Edge Palette_ if there is not one already.
* Inside that palette, create either a _Source Edge Reconnection Tool_ or _Target Edge Reconnection Tool_ (or both if you want to support both cases).
* Inside the body of the tool, you have access to several variables representing the semantic source and target of the edge, before and after the requested reconnect:
  * `edgeView`: the _Edge_ object itslef (usually only needed for very specific scenarios)
  * `edgeSemanticElement`: the semantic element of the edge, if it's a _Domain Based Edge_
  * `semanticReconnectionSource`: the semantic element at the end that is reconnected before the reconnection
  * `reconnectionSourceView`: the graphical element at the end that is reconnected before the reconnection
  * `semanticReconnectionTarget`: the semantic element at the end that is reconnected after the reconnection
  * `reconnectionTargetView`: the graphical element at the end that is reconnected after the reconnection
  * `otherEnd`: the graphical element at the end of the edge that is not reconnected
  * `semanticOtherEnd`: the semantic element at the end of the edge that is not reconnected
